app/core/Router.coffee

```diff
@@ -104,125 +132,146 @@ module.exports = class CocoRouter extends Backbone.Router
+    'teachers/resources': go({path: 'teachers/ResourceHubView', load: -> `import(/* webpackChunkName: "teachers" */ 'views/teachers/ResourceHubView')`}, { redirectStudents: true })
+    'teachers/resources/ap-cs-principles': go({path: 'teachers/ApCsPrinciplesView', load: -> `import(/* webpackChunkName: "teachers" */ 'views/teachers/ApCsPrinciplesView')`}, { redirectStudents: true })
+    'teachers/resources/:name': go({path: 'teachers/MarkdownResourceView', load: -> `import(/* webpackChunkName: "teachers" */ 'views/teachers/MarkdownResourceView')`}, { redirectStudents: true })
     'teachers/signup': ->
       return @routeDirectly('teachers/CreateTeacherAccountView', []) if me.isAnonymous()
       return @navigate('/students', {trigger: true, replace: true}) if me.isStudent() and not me.isAdmin()
       @navigate('/teachers/update-account', {trigger: true, replace: true})
-    'teachers/starter-licenses': go('teachers/StarterLicenseUpsellView', { redirectStudents: true, teachersOnly: true })
+    'teachers/starter-licenses': go({path: 'teachers/StarterLicenseUpsellView', load: -> `import(/* webpackChunkName: "teachers" */ 'views/teachers/StarterLicenseUpsellView')`}, { redirectStudents: true, teachersOnly: true })
     'teachers/update-account': ->
       return @navigate('/teachers/signup', {trigger: true, replace: true}) if me.isAnonymous()
       return @navigate('/students', {trigger: true, replace: true}) if me.isStudent() and not me.isAdmin()
       @routeDirectly('teachers/ConvertToTeacherAccountView', [])
+    'apcsp(/*subpath)': go({path: 'teachers/DynamicAPCSPView', load: -> `import(/* webpackChunkName: "teachers" */ 'views/teachers/DynamicAPCSPView')`})
 
-    'test(/*subpath)': go('TestView')
+    # TestView chunk
+    'test(/*subpath)': go({path: 'TestView', load: -> `import(/* webpackChunkName: "TestView" */ 'views/TestView')`,})
 
-    'user/:slugOrID': go('user/MainUserView')
-    'user/:userID/verify/:verificationCode': go('user/EmailVerifiedView')
+    'identify': go({path: 'user/IdentifyView', load: -> `import(/* webpackChunkName: "user" */ 'views/user/IdentifyView')` })
+    'user/:slugOrID': go({path: 'user/MainUserView', load: -> `import(/* webpackChunkName: "user" */ 'views/user/MainUserView')` })
+    'user/:userID/verify/:verificationCode': go({path: 'user/EmailVerifiedView', load: -> `import(/* webpackChunkName: "user" */ 'views/user/EmailVerifiedView')` })
 
     '*name/': 'removeTrailingSlash'
-    '*name': go('NotFoundView')
+    
+    # NotFoundView chunk
+    '*name': go({path: 'NotFoundView', load: -> `import(/* webpackChunkName: "NotFoundView" */ 'views/NotFoundView')`})
 
   routeToServer: (e) ->
     window.location.href = window.location.href
 
   removeTrailingSlash: (e) ->
     @navigate e, {trigger: true}
+    
+  loadPath: (path) ->
+    path = normalizeViewRoute(path)
+    allRoutes = _.assign({}, _.values(@routes), _.values(sharedViews))
```

Comment:
Can we calculate `allRoutes` once instead of every `routeDirectly` call?  It doesn't seem to change per-call.